_security/fips.md

```diff
@@ -0,0 +1,179 @@
+
+# Use system PKCS11 trust store with specific provider
+./bin/opensearch-fips-demo-installer system --pkcs11-provider YourPKCS11-Provider
+```
+{% include copy.html %}
+
+### Configuration output
+
+After running the FIPS demo installer, the following properties are added to your `jvm.options` file:
+
+```
+################################################################
+## Start OpenSearch FIPS Demo Configuration
+## WARNING: revise all the lines below before you go into production
+################################################################
+
+-Djavax.net.ssl.trustStore=/path/to/opensearch/config/opensearch-fips-truststore.bcfks
+-Djavax.net.ssl.trustStorePassword=<your-password>
+-Djavax.net.ssl.trustStoreType=BCFKS
+-Djavax.net.ssl.trustStoreProvider=BCFIPS
+################################################################
+```
+
+These properties configure the JVM to use the FIPS-compliant trust store for all SSL/TLS connections if no other trust store is defined.
+
+## Troubleshooting FIPS
+
+This section covers common issues when running OpenSearch in FIPS mode.
+
+### Trust store type not specified
+
+```
+Trust store type must be specified using the '-Djavax.net.ssl.trustStoreType' JVM option. Accepted values are PKCS11 and BCFKS.
+```
+
+This error indicates that the FIPS trust store configuration is incomplete or missing from `jvm.options`. To resolve this issue:
+
+- Verify that you have run the FIPS demo installer successfully.
+- Check that `jvm.options` contains the FIPS trust store configuration block.
+
+### Trust store file not found
+
+If you see an error indicating the trust store file cannot be found, verify that:
+
+- The path in `jvm.options` is correct and absolute.
+- The trust store file exists at the specified location.
+- OpenSearch has read permissions for the trust store file.
+
+### Certificate conversion failures
+
+Some certificates in the JVM default trust store may not be compatible with BCFKS format. The installer will report how many certificates were successfully converted. Review the output to ensure critical certificates were converted successfully.
+
+### Keystore password too weak for FIPS mode
+
+If OpenSearch fails to start with the error:
+
+```
+org.bouncycastle.crypto.fips.FipsUnapprovedOperationError: password must be at least 112 bits
+```
+
+This error occurs when the [OpenSearch keystore]({{site.url}}{{site.baseurl}}/install-and-configure/configuring-opensearch/security-settings/#password-protection) `$OPENSEARCH_HOME/config/opensearch.keystore` has a password that does not meet FIPS requirements. In FIPS mode, Bouncy Castle enforces a minimum password strength of 112 bits, which is approximately 14 characters.
+
+Because FIPS mode is already active, the `opensearch-keystore passwd` command will not accept the existing weak password. The workaround is to recreate the keystore:
+
+```bash
+# List existing secrets for backup (if needed)
```

Comment:
Could you split these into individual code blocks with their own copy buttons? Make the comments intro sentences for the code blocks. For example:

List existing secrets for backup (if needed):

```
./bin/opensearch-keystore list
```